cmd/cluster/create/runner.go

```diff
@@ -19,8 +19,7 @@ import (
 
 const (
 	envE2EKubeconfig = "E2E_KUBECONFIG"
-	listVersionsURL  = "https://registry.hub.docker.com/v1/repositories/kindest/node/tags"
-	nodeImage        = "kindest/node"
+	listVersionsURL  = "https://quay.io/api/v1/repository/giantswarm/kind-node/tag/"
```

Comment:
That constant should be in `flag.go` as it is not used here at all as it seems. Further, how much sense does it make to make this configurable? It looks like it is heavily dependent on the image to be used anyway.

Reply:
`listVersionsURL` is used in `runner.go` `listVersions` function, and only there. It's not a flag, just a constant, so that ugly literal can be outside of the function logic.

Reply:
I see, I confused it with `flagListVersions`.

Reply:
With this PR `listVersions` function, in addition to being used for executing `e2ectl cluster create` command when `list-versions` flag is provided, is also used for determining default value of `version` flag. Previously, default `version` was determined from kind library defaults - problem is we cannot guarantee that such default would be valid anymore, since it could happen that we updated kind libary dependency but didn't retag version it defaults to.

PR doesn't address corner case if one provides `image` flag value via CLI, to say `kindest/node` and especially if some other custom image is used - then version probably shouldn't default to anything and validation should check that version is also defined when version. But we aren't building general purpose tool, it's just for internal needs, `image` flag default should be most common scenario and also using `kindest/node` should be safe as IMO it's should be safe assumption that retagged images will always be just a subset of upstream images so versions resolved from quay should be found on docker hub as well.